flink-table/flink-table-api-java/src/main/java/org/apache/flink/table/api/internal/TableEnvironmentImpl.java

```diff
@@ -1600,23 +1619,38 @@ private Object[][] buildTableColumns(ResolvedSchema schema) {
                                                             "PRI(%s)",
                                                             String.join(", ", columns))));
                         });
-
+        boolean nonComments = isSchemaNonColumnComments(schema);
         return schema.getColumns().stream()
                 .map(
                         (c) -> {
                             final LogicalType logicalType = c.getDataType().getLogicalType();
+                            if (nonComments) {
+                                return new Object[] {
+                                    c.getName(),
+                                    logicalType.copy(true).asSummaryString(),
+                                    logicalType.isNullable(),
+                                    fieldToPrimaryKey.getOrDefault(c.getName(), null),
+                                    c.explainExtras().orElse(null),
+                                    fieldToWatermark.getOrDefault(c.getName(), null)
+                                };
+                            }
                             return new Object[] {
                                 c.getName(),
                                 logicalType.copy(true).asSummaryString(),
                                 logicalType.isNullable(),
                                 fieldToPrimaryKey.getOrDefault(c.getName(), null),
                                 c.explainExtras().orElse(null),
-                                fieldToWatermark.getOrDefault(c.getName(), null)
+                                fieldToWatermark.getOrDefault(c.getName(), null),
+                                c.getComment().orElse(null)
                             };
                         })
                 .toArray(Object[][]::new);
     }
 
+    private boolean isSchemaNonColumnComments(ResolvedSchema schema) {
+        return schema.getColumns().stream().noneMatch(col -> col.getComment().isPresent());
```

Comment:
Minor: `schema.getColumns().stream().map(Column::getComment).noneMatch(Optional::isPresent);`
This is just a matter of personal code style.

Reply:
done :)